api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -224,6 +227,17 @@ public void purgeObs(Obs obs, boolean cascade) throws APIException {
 	}
 	
 	/**
+	 * @see org.openmrs.api.ObsService#purgeObs(org.openmrs.Obs, boolean)
+	 */
+	public void purgeObs(Obs obs, boolean cascade) throws APIException {
+		if (purgeComplexData(obs) == false) {
```

Comment:
How about just changing purgeComplexData() to return true if the complex obs is just not found. But always return false if something else went wrong (Failure to actually delete an existing complex obs, e.g due to permissions, etc)